[REFACTOR] Handle nullable on arguments extra

karthyks opened the pull request:
<!-- INSTRUCTION: Keep the line below to notify all core developers about this new PR -->
@RocketChat/android

<!-- INSTRUCTION: Inform the issue number that this PR closes, or remove the line below -->
Closes #2094

#### Changes: [Refactor app module]
1. handle nullable on arguments extra
2. format code to match kotlin guidelines
3. remove redundant underscores on function destructuring
4. handle deprecated calls on lower android versions
#### Screenshots or GIF for the change:

karthik-mtap committed:
handle nullable on arguments extra
format code to match kotlin guidelines
remove redundant underscores on function destructuring
handle deprecated calls on lower android versions

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=2093) <br/>All committers have signed the CLA.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=2093) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=2093) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat.Android?pullRequest=2093) it.</sub>

GOVINDDIXIT commented:
@karthyks Pls create an issue regarding this PR and mention it in the PR description

karthyks commented:
> @karthyks Pls create an issue regarding this PR and mention it in the PR description

Thanks for the info. I added the issue number corresponding to this PR.

philipbrito commented on app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt:
```diff
@@ -58,10 +58,8 @@ class LoginFragment : Fragment(), LoginView {
-        val bundle = arguments
-        if (bundle != null) {
-            serverName = bundle.getString(SERVER_NAME)
+        arguments?.run {
```
💯

philipbrito commented on app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt:
```diff
@@ -52,14 +52,10 @@ class RegisterUsernameFragment : Fragment(), RegisterUsernameView {
+        arguments?.run {
+            userId = getString(BUNDLE_USER_ID, "")
+            authToken = getString(BUNDLE_AUTH_TOKEN, "")
+        } ?: requireNotNull(arguments) { "no arguments supplied when the fragment was instantiated" }
```
👍

philipbrito commented on app/src/main/java/chat/rocket/android/chatroom/adapter/ActionsListAdapter.kt:
```diff
@@ -12,7 +12,10 @@ import com.facebook.drawee.backends.pipeline.Fresco
 import timber.log.Timber
 
-class ActionsListAdapter(actions: List<Action>, var actionAttachmentOnClickListener: ActionAttachmentOnClickListener) : RecyclerView.Adapter<ActionsListAdapter.ViewHolder>() {
+class ActionsListAdapter(
```
👍

philipbrito commented on app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt:
```diff
@@ -174,12 +174,12 @@ class ChatRoomAdapter(
         if (index > -1) {
             dataSet[index] = message
-            dataSet.forEachIndexed { index, viewModel ->
+            dataSet.forEachIndexed { ind, viewModel ->
```
👍

philipbrito commented on app/src/main/java/chat/rocket/android/chatroom/adapter/RoomSuggestionsAdapter.kt:
```diff
@@ -24,8 +24,8 @@ class RoomSuggestionsAdapter : SuggestionsAdapter<RoomSuggestionsViewHolder>("#"
             with(itemView) {
-                val fullname = itemView.findViewById<TextView>(R.id.text_fullname)
-                val name = itemView.findViewById<TextView>(R.id.text_name)
+                val fullname = findViewById<TextView>(R.id.text_fullname)
```
💯

philipbrito commented on app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomActivity.kt:
```diff
@@ -81,44 +81,46 @@ class ChatRoomActivity : AppCompatActivity(), HasSupportFragmentInjector {
 
-        val chatRoomId = intent.getStringExtra(INTENT_CHAT_ROOM_ID)
-        requireNotNull(chatRoomId) { "no chat_room_id provided in Intent extras" }
+        with(intent) {
```
💯

philipbrito commented on app/src/main/java/chat/rocket/android/chatroom/ui/Dialog.kt:
```diff
@@ -25,27 +24,22 @@ fun ChatRoomFragment.showFileAttachmentDialog(uri: Uri) {
+                                .asBitmap()
+                                .load(uri)
+                                .fitCenter()
+                                .into(object : ImageViewTarget<Bitmap>(imagePreview) {
```
💯

philipbrito commented on app/src/main/java/chat/rocket/android/chatrooms/adapter/RoomViewHolder.kt:
```diff
@@ -53,14 +56,14 @@ class RoomViewHolder(itemView: View, private val listener: (RoomUiModel) -> Unit
-                text_chat_name.setTextAppearance(context, R.style.ChatList_ChatName_Unread_TextView)
-                text_timestamp.setTextAppearance(context, R.style.ChatList_Timestamp_Unread_TextView)
-                text_last_message.setTextAppearance(context, R.style.ChatList_LastMessage_Unread_TextView)
+                text_chat_name.setTextViewAppearance(context, R.style.ChatList_ChatName_Unread_TextView)
```
👍

philipbrito commented on app/src/main/java/chat/rocket/android/helper/ImageHelper.kt:
```diff
@@ -51,25 +51,27 @@ object ImageHelper {
-                it.typeface = Typeface.DEFAULT_BOLD
-                it.setPadding(pad)
+            val titleTextView = TextView(context).also { tv ->
+                with(tv) {
```
👍

philipbrito commented on app/src/main/java/chat/rocket/android/members/adapter/MembersAdapter.kt:
```diff
@@ -10,8 +10,9 @@ import chat.rocket.android.util.extensions.inflate
-class MembersAdapter(private val listener: (MemberUiModel) -> Unit) :
-    RecyclerView.Adapter<MembersAdapter.ViewHolder>() {
+class MembersAdapter(
+        private val listener: (MemberUiModel) -> Unit
```
🛑 Wrong indentation.

philipbrito commented on app/src/main/java/chat/rocket/android/createchannel/ui/CreateChannelFragment.kt:
```diff
@@ -1,11 +1,7 @@
-import android.view.MenuItem
-import android.view.View
-import android.view.ViewGroup
+import android.view.*
```
🛑 Remove the wildcard import.

philipbrito commented on app/src/main/java/chat/rocket/android/members/ui/MembersFragment.kt:
```diff
@@ -80,7 +77,7 @@ class MembersFragment : Fragment(), MembersView {
             if (adapter.itemCount == 0) {
                 adapter.prependData(dataSet)
-                if (dataSet.size >= 59) { // TODO Check why the API retorns the specified count -1
+                if (dataSet.size >= 59) { // TODO Check why the API returns the specified count -1
```
Good 👁 
👍

philipbrito commented on app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt:
```diff
@@ -96,9 +96,13 @@ class ProfileFragment : Fragment(), ProfileView, ActionMode.Callback {
         if (resultData != null && resultCode == Activity.RESULT_OK) {
             if (requestCode == REQUEST_CODE_FOR_PERFORM_SAF) {
-                presenter.updateAvatar(resultData.data)
+                resultData.data?.let {
```
Thanks for checking against nullability here.
👍

philipbrito commented on app/src/main/java/chat/rocket/android/push/PushManager.kt:
```diff
@@ -295,15 +298,24 @@ class PushManager @Inject constructor(
 
-    private fun getContentIntent(context: Context, notificationId: Int, pushMessage: PushMessage, grouped: Boolean = false): PendingIntent {
+    private fun getContentIntent(
+            context: Context,
```
🛑 Wrong indentation.

philipbrito commented on app/src/main/java/chat/rocket/android/push/PushManager.kt:
```diff
@@ -230,7 +230,10 @@ class PushManager @Inject constructor(
     @RequiresApi(Build.VERSION_CODES.O)
-    private fun createBaseNotificationBuilder(pushMessage: PushMessage, grouped: Boolean = false): NotificationCompat.Builder {
+    private fun createBaseNotificationBuilder(
+            pushMessage: PushMessage,
```
🛑 Wrong indentation.

philipbrito commented on app/src/main/java/chat/rocket/android/push/PushManager.kt:
```diff
@@ -383,12 +395,12 @@ data class PushMessage(
 ) : Parcelable {
 
     constructor(parcel: Parcel) : this(
+        parcel.readString()!!,
```
🛑 Not so good using `!!`

karthyks replied:
The nullable `parcel.readString()` was ignored before. Now I handled with elvis operator which returns empty if null.

philipbrito commented on app/src/main/java/chat/rocket/android/util/extensions/TextView.kt:
```diff
@@ -0,0 +1,13 @@
+import android.os.Build
+import android.widget.TextView
+
+fun TextView.setTextViewAppearance(context: Context, style: Int) {
```
🛑 It's better to add this extension function on Widget.kt instead of creating a TextView.kt.

philipbrito reviewed (changes requested):
Just need to fixes some things here before the merge.
Could you please take a look at the comments starting with a 🛑 icon?

Thank you very much for sending this PR.

karthik-mtap committed:
fix wildcard imports
fix indentation
handle nullable in PushManager parcelable methods
move textview extension in widget file

karthyks commented:
Are there any problem with CircleCI? The tests were passing on my previous commit, now its failing even though I haven't changed anything major. 

`build foss apk` and `build play apk` are passing when I am running the build locally.

philipbrito reviewed (approved):
Thank you so much for this PR @karthyks!

philipbrito commented:
@karthyks I had to revert this PR here. It was my fault and I didn't see that you wanted to merge it with our `master` branch.

Please, as pointed on the README.md use always the `develop` branch.

_"This repository contains all the code related to the Android native application of Rocket.Chat. To send new pull-requests, _always use the branch develop as base_ and open an issue with the description of what you want/need to accomplish, if the issue wasn't created yet."_
